synapse/storage/database.py

```diff
@@ -667,7 +667,8 @@ def new_transaction(
                 )
         # also check variables referenced in func's closure
         if inspect.isfunction(func):
-            f = cast(types.FunctionType, func)
+            # Keep the cast for now---it helps PyCharm to understand what `func` is.
```

Comment:
I guess we would need this until PyCharm had better understanding of `inspect.isfunction` or something?

Reply:
Yeah. There's a generic mechanism for this kind of pattern (TypeGuard). I'm not sure how well IntelliJ understands it though.